### ISSUE TITLE
p2p connection issue

### ISSUE THREAD

**Issue by ebuchman** [tendermint/tendermint#1400]
Not exactly sure what's going on here:

```
I[04-01|00:12:14.409] Will dial address                            module=p2p addr=fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38@88.99.33.175:46656
I[04-01|00:12:14.409] Dialing peer                                 module=p2p address=fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38@88.99.33.175:46656
I[04-01|00:12:15.056] Successful handshake with peer               module=p2p peer=88.99.33.175:46656 peerNodeInfo="NodeInfo{pk: {PubKeyEd25519{D165E45543660299A42761F09CCDEFBC65CAD242489315D4BA3ECF3257CC44B3}}, moniker: public.cosmos.valid.st, network: test-chain-C1NpdI [listen 88.99.33.175:46656], version: 0.17.1 ([wire_version=0.7.3 p2p_version=0.5.0 consensus_version=v1/0.2.2 rpc_version=0.7.0/3 tx_index=on rpc_addr=tcp://0.0.0.0:46657])}"
I[04-01|00:12:15.056] Starting Peer                                module=p2p peer=88.99.33.175:46656 impl="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
I[04-01|00:12:15.056] Starting MConnection                         module=p2p peer=88.99.33.175:46656 impl=MConn{88.99.33.175:46656}
E[04-01|00:12:15.056] Connection failed @ recvRoutine (reading byte) module=p2p peer=88.99.33.175:46656 conn=MConn{88.99.33.175:46656} err=EOF
I[04-01|00:12:15.056] Stopping MConnection                         module=p2p peer=88.99.33.175:46656 impl=MConn{88.99.33.175:46656}
E[04-01|00:12:15.056] Stopping peer for error                      module=p2p peer="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}" err=EOF
I[04-01|00:12:15.056] Stopping Peer                                module=p2p peer=88.99.33.175:46656 impl="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
D[04-01|00:12:15.056] Stopping MConnection (ignoring: already stopped) module=p2p peer=88.99.33.175:46656 impl=MConn{88.99.33.175:46656}
I[04-01|00:12:15.056] Added peer                                   module=p2p peer="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
I[04-01|00:12:15.057] Stopping queryMaj23Routine for peer          module=consensus peer="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
I[04-01|00:12:15.057] Stopping gossipDataRoutine for peer          module=consensus peer="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
D[04-01|00:12:15.057] Send Pong                                    module=p2p peer=88.99.33.175:46656
I[04-01|00:12:15.057] Stopping gossipVotesRoutine for peer         module=consensus peer="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
D[04-01|00:12:15.057] Flush                                        module=p2p peer=88.99.33.175:46656 conn=MConn{88.99.33.175:46656}
E[04-01|00:12:15.057] MConnection flush failed                     module=p2p peer=88.99.33.175:46656 err="write tcp 159.89.121.174:50418->88.99.33.175:46656: use of closed network connection"
D[04-01|00:12:15.407] Consensus ticker                             module=blockchain numPending=1000 total=1000 outbound=1 inbound=0
D[04-01|00:12:15.408] Blockpool has no peers                       module=blockchain 
I[04-01|00:12:15.876] Dialing peer                                 module=p2p address=fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38@88.99.33.175:46656
D[04-01|00:12:16.407] Consensus ticker                             module=blockchain numPending=1000 total=1000 outbound=1 inbound=0
D[04-01|00:12:16.408] Blockpool has no peers                       module=blockchain 
E[04-01|00:12:16.509] Failed to add peer                           module=p2p address=fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38@88.99.33.175:46656 err="Duplicate peer"
E[04-01|00:12:16.510] Error dialing peer                           module=p2p err="Duplicate peer"
```

Notes:
- Connection fails right away with `EOF`
- We try to dial again and get "Duplicate peer" error
- Later when we call ensurePeers, we send message to this peer:

```
I[04-01|00:12:44.409] Ensure peers                                 module=p2p numOutPeers=1 numInPeers=0 numDialing=0 numToDial=9
I[04-01|00:12:44.410] We need more addresses. Sending pexRequest to random peer module=p2p peer="Peer{MConn{88.99.33.175:46656} fa9b3b2d98cd9b16deb4d08473e4eb80fc4c9b38 out}"
```

But the block pool continues to think we have no peers.

Note the peer shows up in the `/net_info`, but with the cur/inst rate at 0 and the idle time ever growing ...

**Comment by xla** [comment]
Closed in favour of #1744

**Comment by njmurarka** [comment]
Noticed the "MConnection flush failed" message was also in the two issues I filed:

#4924 and #4925
